driver-sync/src/test/functional/com/mongodb/client/ClientSideEncryptionCorpusTest.java

```diff
@@ -0,0 +1,292 @@
+            String method = corpusEncryptedActual.getDocument(field).getString("method").getValue();
+            // TODO: unused
+            String identifier = corpusEncryptedActual.getDocument(field).getString("identifier").getValue();
+            boolean allowed = corpusEncryptedActual.getDocument(field).getBoolean("allowed").getValue();
+            BsonValue value = corpusEncryptedActual.getDocument(field).get("value");
+
+            // All deterministic fields are an exact match.
+            if (algo.equals("det")) {
+                deterministicCheck(kms, type, corpusEncryptedExpected);
+            }
+
+            if (algo.equals("rand") && !method.equals("prohibited")) {
+                randomCheck(value, field, corpusEncryptedExpected);
+            }
+
+            if (allowed) {
+                BsonValue decrypted = clientEncryption.decrypt(value.asBinary());
+                assertEquals("Values should be equal for field " + field, corpus.getDocument(field).get("value"), decrypted);
+            } else {
+                assertEquals("Values should be equal for field " + field, corpus.getDocument(field).get("value"),
+                        corpusEncryptedActual.getDocument(field).get("value"));
+            }
+        }
+    }
+
+    // Check that all values of document that are deterministic with the same kms + type match value
+    private static void deterministicCheck(final String matchKms, final String matchType, final BsonDocument document) {
+        for (String field : document.keySet()) {
+            if (field.equals("_id") || field.equals("altname_aws") || field.equals("altname_local")) {
+                continue;
+            }
+
+            BsonDocument subDocument = document.getDocument(field);
+
+            String kms = subDocument.getString("kms").getValue();
+            String type = subDocument.getString("type").getValue();
+            String algo = subDocument.getString("algo").getValue();
+            BsonValue value = subDocument.get("value");
+
+            if (kms.equals(matchKms) && type.equals(matchType) && algo.equals("det")) {
+                assertEquals(value, value);
+            }
+        }
+    }
+
+    private static void randomCheck(final BsonValue value, final String excludeField, final BsonDocument document) {
+        for (String field : document.keySet()) {
+            if (field.equals(excludeField)) {
+                continue;
+            }
+            assertNotEquals(document.get(field), value);
+        }
+    }
+
+    private static BsonDocument bsonDocumentFromPath(final String path) throws IOException, URISyntaxException {
+        return getTestDocument(new File(ClientSideEncryptionTest.class
+                .getResource("/client-side-encryption-corpus/" + path).toURI()));
+    }
+
+    @Parameterized.Parameters(name = "useLocalSchema: {0}")
+    public static Collection<Object[]> data() {
+        return Arrays.asList(new Object[]{true}, new Object[]{false});
```

Comment:
:)

Reply:
Is there no better way to to this with JUnit?